test/unit/queue/util/createConsumer-test.js

```diff
@@ -193,6 +196,87 @@ test('should acknowledge a message if consumer receives a message', async (t) =>
   t.pass()
 })
 
+test('should receive a message with converted data in onMessage', async (t) => {
+  const { sandbox, connection } = t.context
+
+  const spy = sandbox.spy()
+
+  const consumer = await createConsumer({
+    amqUrl: TEST_AMQ_URL,
+    logger: console,
+    onMessage: sandbox.spy(),
+    consumerOptions: {
+      queueName: TEST_QUEUE_NAME,
+      connection
+    }
+  })
+
+  const message = {
+    content: encodedMessage
+  }
+
+  consumer.emit('message', message)
+
+  await tri(async () => {
+    let calledWith = spy.calledWith(message)
+    if (calledWith) {
+      const receivedMessage = spy.firstCall.args[0]
+      const receivedMessageCleaned = testMessage.clean()
+
+      t.deepEqual(receivedMessageCleaned, testMessage.clean())
+      t.is(receivedMessage.getData().getCompare(), 'abc123')
+      t.deepEqual(receivedMessage.getData().clean(), testMessageCleaned.data)
+    }
+    return calledWith
```

Comment:
This is probably why the test was passing. Doesn't `tri` reattempt if the promise rejects? You should throw if `calledWith` is false.